File: backend_samples/library/dotnet/AzureAIVisionFaceDeviceAttestation/src/AttestationContext.cs
using Azure.AI.Vision.Face.DeviceAttestation.Configuration;
using Azure.AI.Vision.Face.DeviceAttestation.Logging;
using Azure.AI.Vision.Face.DeviceAttestation.Storage;

namespace Azure.AI.Vision.Face.DeviceAttestation;

/// <summary>
/// The injected dependencies threaded through the handlers and platform
/// verifiers: runtime config, the persistent store, and the telemetry sink.
/// Replaces the npm library's mutable module-global config/logger with explicit
/// dependency injection.
/// </summary>
internal sealed class AttestationContext
{
    public required AttestationConfig Config { get; init; }
    public required IClusterStore Store { get; init; }
    public required IAttestationLogger Logger { get; init; }
}

File: backend_samples/library/java/AzureAIVisionFaceDeviceAttestation/src/main/java/com/azure/ai/vision/face/deviceattestation/AttestationContext.java
package com.azure.ai.vision.face.deviceattestation;

import com.azure.ai.vision.face.deviceattestation.config.AttestationConfig;
import com.azure.ai.vision.face.deviceattestation.logging.AttestationLogger;
import com.azure.ai.vision.face.deviceattestation.store.ClusterStore;

/**
 * The injected dependencies threaded through the handlers and platform
 * verifiers: runtime config, the persistent store, and the telemetry sink.
 * Replaces the npm library's mutable module-global config/logger with explicit
 * dependency injection.
 */
public final class AttestationContext {

    public final AttestationConfig config;
    public final ClusterStore store;
    public final AttestationLogger logger;

    public AttestationContext(AttestationConfig config, ClusterStore store, AttestationLogger logger) {
        this.config = config;
        this.store = store;
        this.logger = logger;
    }
}
